jdaviz/core/tools.py

```diff
@@ -32,6 +33,28 @@
 BqplotYRangeMode.icon = os.path.join(ICON_DIR, 'select_y.svg')
 
 
+@viewer_tool
+class BqplotXRangeModeUnitAware(BqplotXRangeMode):
+    icon = os.path.join(ICON_DIR, 'select_x.svg')
+    tool_id = 'jdaviz:xrange'
+    action_text = 'X range ROI'
+    tool_tip = 'Select a range of x values'
+
+    def update_selection(self, *args):
+        with self.viewer._output_widget:
+            if self.interact.selected is not None:
+                x = self.interact.selected
+                if x is not None and len(x):
+                    x = (x * self.viewer.xunit).to_value(self.viewer._native_xunit, equivalencies=self.viewer._x_equivalencies)
+                    roi = RangeROI(min=min(x), max=max(x), orientation='x')
+                    self.viewer.apply_roi(roi)
+                    mark = self.viewer.figure.marks[-1]
+                    mark._x_equivalencies = self.viewer._x_equivalencies
+                    mark._y_equivalencies = self.viewer._y_equivalencies
+                    mark.set_native_units(self.viewer._native_xunit, self.viewer._native_yunit)
+                    mark.set_display_units(self.viewer.xunit, self.viewer.yunit)
```

Comment:
this is ugly - this is essentially a copy of the logic in glue-jupyter, but converting the values on `x` set to `RangeROI` and then immediately setting the units information on the created `Line` mark (which is _assumed_ to be appended to the end of the `viewer.figure.marks` list).  Buuuuut, it works 🤷